hack/src/verify_yaml.rs

```diff
@@ -0,0 +1,117 @@
+use cincinnati::plugins::internal::openshift_secondary_metadata_parser::plugin;
+use regex::Regex;
+use semver::Version;
+use serde::de::DeserializeOwned;
+use serde_yaml;
+use std::collections::HashSet;
+use std::path::PathBuf;
+use tokio;
+
+use anyhow::Context;
+use anyhow::Result as Fallible;
+
+pub async fn run() -> Fallible<HashSet<Version>> {
+  let data_dir = PathBuf::from("..");
+  let extension_re = Regex::new("ya+ml")?;
+  // Collect a list of mentioned versions
+  let mut found_versions: HashSet<Version> = HashSet::new();
+
+  println!("Verifying blocked edge files are valid");
+  let blocked_edge_path = data_dir.join(plugin::BLOCKED_EDGES_DIR).canonicalize()?;
+  let blocked_edge_vec =
+    walk_files::<plugin::graph_data_model::BlockedEdge>(&blocked_edge_path, &extension_re).await?;
+  for v in blocked_edge_vec.iter() {
+    found_versions.insert(v.to.clone());
+  }
+
+  println!("Verifying channel files are valid");
+  let channel_path = data_dir.join(plugin::CHANNELS_DIR).canonicalize()?;
+  let channels_vec =
+    walk_files::<plugin::graph_data_model::Channel>(&channel_path, &extension_re).await?;
+  for c in channels_vec.iter() {
+    for v in c.versions.iter() {
+      found_versions.insert(v.clone());
+    }
+  }
+
+  Ok(found_versions)
+}
+
+#[derive(Debug)]
+enum ProcessingError {
```

Comment:
I think that we don't actually need the `ProcessingError` enum if we just coerce everything to a string at the end. It would in fact be enough if you just appended the error messages to a mutable string throughout the processing. I don't feel strongly about it though so I'm not opposed to keeping this either.

Reply:
I think it would be useful. We might want to add more error types later - i.e. signature check - with customized Display

Reply:
Adding more types doesn't make it more useful if we still only use it for string coercion :wink: What would convince me if we pass back the error vector to the caller, which would allow it to use the type system to figure out error conditions.

Reply:
>Adding more types doesn't make it more useful 

The idea is to leave formatting errors to Debug impl of ProcessingError. Other tests may extend that struct later.

> What would convince me if we pass back the error vector to the caller

`verify_yamls::run` may throw generic errors, e.g. directory not readable or "blocked-edges" dir cannot be canonicalized.